pkg/kubelet/apis/config/validation/validation.go

```diff
@@ -309,5 +311,15 @@ func ValidateKubeletConfiguration(kc *kubeletconfig.KubeletConfiguration, featur
 		}
 	}
 
+	if kc.SingleProcessOOMKill {
+		if runtime.GOOS == "windows" {
+			allErrors = append(allErrors, fmt.Errorf("invalid configuration: singleProcessOOMKill is not supported on Windows"))
+		}
+
+		if !kubeletutil.IsCgroup2UnifiedMode() {
+			allErrors = append(allErrors, fmt.Errorf("invalid configuration: singleProcessOOMKill requires cgroup v2 unified mode"))
```

Comment:
This conflicts with your cgroup v1 defaulting, which is "true". We should allow setting this to true explicitly.